### ISSUE TITLE
Issue 2: Add Cypress for End-to-End (E2E) Testing

### ISSUE THREAD

**Issue by lcrojano** [lcrojano/Giphy_Explorer#51]
Description:
Introduce Cypress to implement end-to-end testing for the user flow, ensuring a smooth and error-free experience across the application.

Acceptance Criteria:

Cypress is installed and configured.
E2E tests .
Integrate E2E tests with CI pipeline.

**Comment by Suvrat1629** [comment]
Hey I would like to work on this issue, could you assign it to me

**Comment by Suvrat1629** [comment]
i would like to work on this issue but i am not able to find the backend folder or any endpoints for login or navigation. Could you help me with that

**Comment by lcrojano** [comment]
Since there is currently no backend, we are working on a folder structure in #64 to setup a well-organized folder structure to facilitate future development. This structure will support Firebase authentication and integration with MongoDB Atlas for any future backend functionality.

a potential structure could be like this
``` 
├──/cypress                # Cypress end-to-end tests
│   ├── /fixtures       # Sample data for tests
│   ├── /e2e # E2E tests
│   └── /support        # Custom commands and configurations for Cypress
├── /tests              # Unit and integration tests
│   ├── /components     # Tests for UI components
│   ├── /domains        # Tests for domain logic
│   ├── /hooks          # Tests for custom hooks
│   └── /services       # Tests for service functions
│
```
It is enough to create the structures and test templates, as later this would grow over this base architecture.

**Comment by Suvrat1629** [comment]
ok great! i am willing to work on this issue could you assign it to me

**Comment by lcrojano** [comment]
closed by #71
